web/static/app/src/controllers/send_controller.js

```
@@ -29,13 +30,102 @@ export default class extends Controller {
+
+    const addressInput = destinationTemplate.querySelector('input[name="destination-address"]')
+    const amountInput = destinationTemplate.querySelector('input[name="destination-amount"]')
+    const sendMaxButton = destinationTemplate.querySelector('button[type="button"]')
+
+    addressInput.setAttribute('data-index', this.destinationCount)
+    amountInput.setAttribute('data-index', this.destinationCount)
+    sendMaxButton.setAttribute('data-index', this.destinationCount)
+
     this.destinationsTarget.appendChild(destinationTemplate)
 
-    if (this.destinationCount() > 1) {
+    this.destinationCount++
+
+    if (this.destinationCount > 1) {
       this.show(this.removeDestinationButtonTarget)
     }
   }
 
+  setMaxAmountForDestination (event) {
+    const targetElement = event.currentTarget
+    const index = parseInt(targetElement.getAttribute('data-index'))
+
+    let amountField
+    this.amountTargets.forEach(el => {
+      if (index === parseInt(el.getAttribute('data-index'))) {
+        amountField = el
+      }
+    })
+    const currentAmount = amountField.value
+    // temporarily set the destination amount field (if empty) to make destination validation pass
+    if (!(parseInt(currentAmount) > 0)) {
+      amountField.value = 1
+    }
+    if (!this.destinationFieldsValid()) {
+      amountField.value = currentAmount
+      return
+    }
+    this.resetChangeOutput()
+    let _this = this
+    // set the destination amount to zero and get the server calculated value
+    amountField.value = 0
+    this.getMaxSendAmount(changeOutputs => {
+      if (!changeOutputs || changeOutputs.length < 1) {
+        amountField.value = currentAmount
+        return
+      }
+      amountField.value = changeOutputs[0].Amount
+      _this.calculateCustomInputsPercentage()
+    }, (errMsg) => {
+      if (errMsg.indexOf('total input amount not enough to cover transaction') > -1) {
```

2 things here
1) If the total input amount is not enough to generate another amount for another address (the "change address"), we should show the user a proper message; perhaps: `Total send amount is already at maximum`
2) This if statement feels like a hack... what if the error message returned is different? There should be a better way to determine this scenario on the server and the server should return the proper error message described above; and the error message should always be shown to the user